tests/test_config.py

```diff
@@ -0,0 +1,163 @@
+import mock
```

Comment:
Use pytest's `monkeypatch` fixture rather than adding a dependency on mock.

Reply:
I don't think monkeypatch works when using the advanced features of mock (inspecting calls to the mocked function etc) unless you want to reimplement this logic in the test (which is ugly IMO). But I'd use pytest-mock instead of the decorators.

Reply:
It looked like monkeypatch would be fine for the uses here, I agree (pytest-)mock is more appropriate for more complex tasks.

Reply:
can also get fancy and only include mock for py2.

```python
try:
    from unittest import mock
except ImportError:
    # py2
    import mock
```

(I also find mock much more useful & less confusing than pytest's monkeypatch in almost all cases)

Reply:
If the only argument is to not use mock b/c it adds a dependency, `TestCreateEngine` below uses mock.patch extensively to inspect call args.  I've looked at the docs for [pytest monkeypatch](https://docs.pytest.org/en/latest/reference.html#_pytest.monkeypatch.MonkeyPatch) and don't believe it's going to support that use case.

So, if we end up using mock for those tests, might as well use it here and keep things consistent.  Also, I agree with @fuhrysteve that the mock API feels a lot cleaner and obvious than what pytest's offering.

I personally don't think we should be afraid of adding dependencies when they provide a benefit and IMO, this is one of those cases.  I also use mock extensively in my day job and will be most productive with it in future FSA updates.